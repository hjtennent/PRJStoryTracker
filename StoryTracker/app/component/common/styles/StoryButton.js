import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  orangeFollowStoryButton: {
    backgroundColor: "#F25E2E",
    borderRadius: 20,
    justifyContent: "center",
    alignItems: "center",
    padding: 10,
  },
  greenFollowStoryButton: {
    backgroundColor: "#4D8E6B",
    borderRadius: 20,
    justifyContent: "center",
    alignItems: "center",
    padding: 10,
  },
  smallButtonText: {
    color: "white",
    fontFamily: "Oxygen-Bold",
    fontSize: 12,
    justifyContent: "center",
  },
  buttonText: {
    color: "white",
    fontFamily: "Oxygen-Bold",
    fontSize: 16,
    justifyContent: "center",
  },
});

export default styles;