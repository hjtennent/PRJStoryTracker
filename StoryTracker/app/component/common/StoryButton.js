import React from 'react';
import styles from "./styles/StoryButton";
import {
  TouchableOpacity, Text
} from 'react-native';

export default StoryButton = (props) => {
  return (
    <TouchableOpacity
      id={'storyButton'}
      style={props.makeHighlight ? styles.orangeFollowStoryButton : styles.greenFollowStoryButton}
      onPress={() => props.onPress()}>
        <Text style={props.smallText ? styles.smallButtonText : styles.buttonText}>{props.text}</Text>
    </TouchableOpacity>
  )
}