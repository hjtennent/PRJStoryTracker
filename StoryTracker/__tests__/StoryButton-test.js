import React from 'react';
import renderer from 'react-test-renderer';
import StoryButton from '../app/component/common/StoryButton';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { render, fireEvent } from '@testing-library/react-native';

configure({ adapter: new Adapter() });

test('StoryButton renders correctly', () => {
  const tree = renderer.create(
      <StoryButton onPress={jest.fn()} makeHighlight={false}
        text={"Text"} smallText={"Small Text"}/>
    ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('onPress function fires', () => {
  const mockFn = jest.fn();
  const { getByText } = render(
    <StoryButton onPress={mockFn} makeHighlight={false} text="Text" 
      smallText="Small Text" />
  );
  fireEvent.press((getByText('Text')))
  expect(mockFn.mock.calls.length).toBe(1);
})

