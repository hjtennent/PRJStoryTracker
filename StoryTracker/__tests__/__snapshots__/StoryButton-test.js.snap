// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoryBox renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#4D8E6B",
      "borderRadius": 20,
      "justifyContent": "center",
      "opacity": 1,
      "padding": 10,
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontFamily": "Oxygen-Bold",
        "fontSize": 12,
        "justifyContent": "center",
      }
    }
  >
    Text
  </Text>
</View>
`;
